### PR TITLE
Bigdl nano: Makes torch function replacement a plugin-like operation

### PR DESCRIPTION
**Pull request by leonardozcm**
For the development of ipex is on going, there are some operations or functions we can not call directly [#4600](https://github.com/intel-analytics/analytics-zoo/pull/4600#discussion_r699773873).
This pr provides a way for us to apply any workaround as an alternative if it fixs the issue temporarily. The replacement takes place whenever ipex accelerator is imported, in another word `use_ipex` is specfied as true in Trainer, but it will not affect default behaviors.

### CODE REVIEW

File: python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import torch
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py:1]: license

File: python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Tensor = torch.Tensor
+torch_nn_functional = torch.nn.functional
+_Reduction = torch.nn._reduction
+
+
+def replace_torch_function(function_name: str, replace_func: Callable):
+    setattr(torch.nn.functional, function_name, replace_func)
+
+
+def workaround_cross_entropy(
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py:14]: We'd better document what's different between this version and the pytorch version.

File: python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            ignore_index=ignore_index,
+            reduce=reduce,
+            reduction=reduction,
+        )
+    if size_average is not None or reduce is not None:
+        reduction = _Reduction.legacy_get_string(size_average, reduce)
+
+    # `log_softmax(input, 1).contiguous()` makes multidimensional tensors continuous in memory
+    # May fix https://github.com/intel/intel-extension-for-pytorch/issues/175#issue-972312586
+    return torch_nn_functional.nll_loss(torch_nn_functional.log_softmax(input, 1).contiguous(),
```
Comment (yangw1234) [inline python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py:40]: Wouldn't it be simpler to just call `input.cpu()` and `target.cpu()` than call `torch_nn_functional.cross_entropy`?

File: python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py (hunk 2 above)
Comment (yangw1234) [reply python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py:14]: You may also need to cite pytorch's license.

File: python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py (hunk 3 above)
Comment (leonardozcm) [reply python/nano/src/bigdl/nano/pytorch/accelerators/ipex_torchfunctional.py:40]: > Wouldn't it be simpler to just call `input.cpu()` and `target.cpu()` than call `torch_nn_functional.cross_entropy`?

Will this leads to some performance loss when move tensor from xpu to cpu or losing optimizations in cross_entropy provided by ipex? I'm not sure, but only adding `contiguous()` makes minimal modifications compared with original version, so I think we can keep this.

**Comment by leonardozcm** [comment]
http://10.112.231.51:18888/job/ZOO-PR-nano-ipex/20/
http://10.112.231.51:18888/job/ZOO-PR-nano-codestyle-test/26/
http://10.112.231.51:18888/job/ZOO-PR-NanoTests/20/

**Comment by yangw1234** [comment]
LGTM

**Comment by jason-dai** [comment]
Please add detailed description (e.g., in the comments) for the functionality.

**Comment by leonardozcm** [comment]
> Please add detailed description (e.g., in the comments) for the functionality.

sure, I will fix it
